Chapter 5 Task 21:

tasiq opened the pull request:
в классе Parallelepiped осталось много лишнего, но я бы хотел это на будущее сохранить

tasiq committed:
Chapter 5 Task 21:
в классе Parallelepiped осталось много лишнего, но я бы хотел это на будущее сохранить

EvgeneZhurov commented on src/main/java/com/walking/intensive/chapter4/task20/Main.java:
```diff
@@ -1,7 +1,29 @@
 public class Main {
     public static void main(String[] args) {
 
+        Point point1 = new Point(-4, -4, 8);
```
Point point1 = new Point(-4, -4, -1);
        Point point2 = new Point(4, 4, 1);
        Sphere sphere = new Sphere(0, 0, 2, 1);
ответ: false
ожидалось: true

tasiq committed:
Chapter 5 Task 21:

Исправлена опечатка в методе isIntersectedAxisZ, где было обращение к MaxX параллелепипеда.

Копипаст зло.

tasiq committed:
Chapter 5 Task 21:

- код перенесен из Task20 перенесен в Task21;
- исправлена опечатка в методе isIntersectedAxisZ, где было обращение к MaxX параллелепипеда.

Копипаст зло.

EvgeneZhurov commented on src/main/java/com/walking/intensive/chapter5/task21/Main.java:
```diff
@@ -3,5 +3,12 @@
 public class Main {
     public static void main(String[] args) {
 
+        Point point1 = new Point(-4, -4, -1);
```
Point point1 = new Point(-4, -4, -4);
        Point point2 = new Point(4, 4, 4);
        Sphere sphere = new Sphere(0, 0, 0, 100);
результат: false
ожидалось: true

tasiq committed:
Chapter 5 Task 21:

- фигуры "шар" и "параллелепипед" - сплошные, любые точки внутри этих фигур являются их частью.
- изменена логика метода isIntersected: если шар полностью внутри ПП или если ПП полностью внутри шара - есть пересечение. Лишние логические проверки метода удалены.
- класс Sphere переименован в Ball.

EvgeneZhurov commented on src/main/java/com/walking/intensive/chapter5/task21/Main.java:
```diff
@@ -3,5 +3,24 @@
 public class Main {
     public static void main(String[] args) {
 
+        Point point1 = new Point(-4, -4, -4);
```
Point point1 = new Point(0, 0, 0);
Point point2 = new Point(1, 1, 1);
Ball ball1 = new Ball(2, 2, 2, 1);
результат true
ожидалось false
